Assets/Mirage/Runtime/NetworkSceneManager.cs

```diff
@@ -182,210 +173,468 @@ internal void OnClientChangeScene(string scenePath, SceneOperation sceneOperatio
+            }
         }
 
         /// <summary>
-        /// This causes the server to switch scenes and sets the ActiveScenePath.
-        /// <para>Clients that connect to this server will automatically switch to this scene. This automatically sets clients to be not-ready. The clients must call Ready() again to participate in the new scene.</para>
+        /// 
         /// </summary>
-        /// <param name="scenePath"></param>
-        /// <param name="operation"></param>
-        /// <exception cref="ArgumentNullException">Scene path was not valid.</exception>
-        public void ChangeServerScene(string scenePath, SceneOperation sceneOperation = SceneOperation.Normal)
+        /// <param name="scene"></param>
+        /// <param name="players"></param>
+        /// <param name="sceneOperation"></param>
+        private void ServerSceneUnLoading(Scene scene, INetworkPlayer[] players, SceneOperation sceneOperation)
         {
-            if (string.IsNullOrEmpty(scenePath))
-            {
-                throw new ArgumentNullException(nameof(scenePath), "ServerChangeScene: " + nameof(scenePath) + " cannot be empty or null");
-            }
+            if(scene.handle == 0)
+                throw new ArgumentNullException(nameof(scene),
+                    "[NetworkSceneManager] - ServerChangeScene: " + nameof(scene) + " cannot be null");
+
+            if (players == null || players.Length == 0)
+                throw new ArgumentNullException(nameof(scene),
+                    "[NetworkSceneManager] - list of player's cannot be null or no players.");
 
-            if (logger.LogEnabled()) logger.Log("ServerChangeScene " + scenePath);
+            if (logger.LogEnabled()) logger.Log("[NetworkSceneManager] - ServerChangeScene " + scene.name);
 
             // Let server prepare for scene change
-            OnServerChangeScene(scenePath, sceneOperation);
+            logger.Log("[NetworkSceneManager] - OnServerChangeScene");
+
+            ServerStartedSceneChange?.Invoke(scene.path, sceneOperation);
 
             if (!Server.LocalClientActive)
-                ApplyOperationAsync(scenePath, sceneOperation).Forget();
+                UnLoadSceneAsync(scene, sceneOperation).Forget();
 
             // notify all clients about the new scene
-            Server.SendToAll(new SceneMessage { scenePath = scenePath, sceneOperation = sceneOperation });
+            if (players.Length > 0)
+                NetworkServer.SendToMany(players,
+                    new SceneMessage { MainActivateScene = scene.path, SceneOperation = sceneOperation });
+        }
+
+        /// <summary>
+        ///     Allows server to fully load in a new scene and override current active scene.
+        /// </summary>
+        /// <param name="scenePath">The full path to the scene file or the name of the scene.</param>
+        public void ServerLoadSceneNormal(string scenePath)
+        {
+            if(!Server && !Server.Active) return;
```

Comment:
this should throw `InvalidOperationException` instead of return.

Reply:
done